File: client-app/src/modules/applications/types/Application.ts
// src/modules/applications/types/Application.ts

export type ApplicationStatus = "Applied" | "Interview" | "Offer" | "Rejected";

export interface Application {
  id: number;
  position: string;
  company: string;
  link?: string | null;
  notes?: string | null;
  status: ApplicationStatus;
  createdAt: string;
  updatedAt: string;
  userId: number;
}

export interface ApplicationCreateDto {
  position: string;
  company: string;
  link?: string | null;
  notes?: string | null;
  status: ApplicationStatus;
}

export interface ApplicationUpdateDto {
  position: string;
  company: string;
  link?: string | null;
  notes?: string | null;
  status: ApplicationStatus;
}

// ✅ Нови enum-и за сортиране и филтриране

export enum SortBy {
  Position = "Position",
  Company = "Company",
  CreatedAt = "CreatedAt",
}

export enum SortDirection {
  Asc = "asc",
  Desc = "desc",
}

// ✅ Query параметри, които ползваме при fetch
export interface ApplicationQueryParams {
  search?: string;
  page?: number;
  pageSize?: number;
  sortBy?: SortBy;
  sortDirection?: SortDirection;
}

File: ApplicationsService/Models/Application.cs
using IdentityService.Models;

namespace ApplicationsService.Models;

public enum ApplicationStatus { Applied, Interview, Offer, Rejected }

public class Application
{
    public int Id { get; set; }
    public string Position { get; set; } = string.Empty;
    public string Company { get; set; } = string.Empty;
    public string? Link { get; set; }
    public string? Notes { get; set; }
    public ApplicationStatus Status { get; set; } = ApplicationStatus.Applied;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public int UserId { get; set; }             // от JWT (sub)
    public User User { get; set; } = null!;
}
